src/common/rbac/evaluator.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/casbin/casbin"
+)
+
+// Evaluator the permissioin evaluator
+type Evaluator interface {
+	// HasPermission returns true when user has action permission for the resource
+	HasPermission(resource Resource, action Action) bool
+}
+
+type rbacEvaluator struct {
+	user     User
+	enforcer *casbin.Enforcer
+	once     sync.Once
+}
+
+func (e *rbacEvaluator) HasPermission(resource Resource, action Action) bool {
+	e.once.Do(func() {
+		e.enforcer = enforcerForUser(e.user)
+	})
+
+	return e.enforcer.Enforce(e.user.GetUserName(), resource.String(), action.String())
+}
+
+// NewEvaluator returns Evaluator for the rbac user
+func NewEvaluator(user User) Evaluator {
```

Comment:
NewRbacEvaluator Or call it `userEvaluator` and `NewUserEvaluator`?

Reply:
Changed